### PR TITLE
[cmake] allow custom OTBR package name and package version

### PR DESCRIPTION
**Pull request by suveshpratapa**
@jwhui

- Was turning OT_DUA off by default for a Thread 1.2 border router intentional? Let us know if you're okay with that change.
- Additionally, is there any reason COAP and COAPS are not turned on for the border router stack by default?

### CODE REVIEW

**Comment by jwhui** [comment]
> * Was turning OT_DUA off by default for a Thread 1.2 border router intentional? Let us know if you're okay with that change.

Yes, disabling DUA by default was intentional since this feature is now optional in Thread 1.2 - see [TESTPLAN-489](https://threadgroup.atlassian.net/browse/TESTPLAN-489).

> * Additionally, is there any reason COAP and COAPS are not turned on for the border router stack by default?

I don't know if disabling COAP(S) features by default was a conscious decision. FWIW, on Linux platform, most users seem to use external CoAP libraries. Are you looking to enable these features primarily for testing purposes?

**Comment by suveshpratapa** [comment]
> Yes, disabling DUA by default was intentional since this feature is now optional in Thread 1.2 - see [TESTPLAN-489](https://threadgroup.atlassian.net/browse/TESTPLAN-489).

Okay, yes, I remember now.  Reverted that change.

> I don't know if disabling COAP(S) features by default was a conscious decision. FWIW, on Linux platform, most users seem to use external CoAP libraries. Are you looking to enable these features primarily for testing purposes?

Yes, we have turned on COAPS primarily for testing (needs a change to the default mbedtls header as well). I figure it couldn't hurt to provide that out-of-the-box capability for those looking to test, since the stack provides it. What do you think?

File: CMakeLists.txt [hunk 1]
```diff
@@ -56,17 +56,24 @@ endif()
 
 add_compile_options(-Wall -Wextra -Werror -Wfatal-errors -Wno-missing-braces)
 
+if(NOT OTBR_NAME)
+    set(OTBR_NAME "\"OPENTHREAD_BR\"")
```
Comment (wgtdkp) [inline CMakeLists.txt:60]:
```suggestion
    set(OTBR_NAME "OPENTHREAD_BR")
```

**Review by wgtdkp — APPROVED** [review]
LGTM 👍

File: CMakeLists.txt (hunk 1 above)
Comment (suveshpratapa) [reply CMakeLists.txt]: Should be fixed now, thanks.

File: third_party/openthread/CMakeLists.txt [hunk 2]
```diff
@@ -71,6 +71,9 @@ if (OT_THREAD_VERSION STREQUAL "1.2")
     set(OT_MLR ON CACHE BOOL "Enable Thread 1.2 MLR by default")
 endif()
 
+set(OT_COAP ON CACHE BOOL "Enable CoAP in OpenThread" FORCE)
```
Comment (bukepo) [inline third_party/openthread/CMakeLists.txt:74]: It seems this disables us from disabling the feature even with `-DOT_COAP=OFF`.
```suggestion
set(OT_COAP ON CACHE BOOL "Enable CoAP in OpenThread")
```

File: third_party/openthread/CMakeLists.txt [hunk 3]
```diff
@@ -71,6 +71,9 @@ if (OT_THREAD_VERSION STREQUAL "1.2")
...
+set(OT_COAPS ON CACHE BOOL "Enable secure CoAP in OpenThread" FORCE)
```
Comment (bukepo) [inline third_party/openthread/CMakeLists.txt:75]:
```suggestion
set(OT_COAPS ON CACHE BOOL "Enable secure CoAP in OpenThread")
```

File: third_party/openthread/CMakeLists.txt (hunk 2 above)
Comment (suveshpratapa) [reply third_party/openthread/CMakeLists.txt]: Fixed! Thanks.

File: third_party/openthread/CMakeLists.txt (hunk 3 above)
Comment (suveshpratapa) [reply third_party/openthread/CMakeLists.txt]: Fixed! Thanks.

**Comment by suveshpratapa** [comment]
@bukepo Is there anything else you'd like to see in this change? I addressed your other comments.

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**